### ISSUE TITLE
Getting rid of missing value labels in the plot and frq output.

### ISSUE THREAD

**Issue by haseebmahmud** [strengejacke/sjPlot#361]
I am having an issue with `sjp.frq` plot. I have imported the data using this command, 

`en_w1 <- read_spss("rel.sav", atomic.to.fac = TRUE)
`
Originally, the frequency distribution from SPSS output is as follows, 
![frequency](https://user-images.githubusercontent.com/544033/40536359-2e1e2e2a-600c-11e8-8bc5-9151c9cccea3.png)

The variable is coded as follows,

![2018-05-25 10_44_09-value labels](https://user-images.githubusercontent.com/544033/40536390-44e36242-600c-11e8-9cf3-6d181cbb602b.png)

And here is the missing value coding where anything below zero is labeled as missing data,

![2018-05-25 10_44_35-missing values](https://user-images.githubusercontent.com/544033/40536406-50067236-600c-11e8-95de-357276bfa8dd.png)

After importing the data, the frq command gives me the following output, 

```
> frq(en_w1$y1_rel1)

# m71: what is your religion (x) <categorical> 
# total N=4315  valid N=4024  mean=3.20  sd=2.58
 
 val                    label  frq raw.prc valid.prc cum.prc
 -88                no answer    0    0.00      0.00    0.00
 -77           not applicable    0    0.00      0.00    0.00
 -66       question not asked    0    0.00      0.00    0.00
 -55            other missing    0    0.00      0.00    0.00
  -1           dropout wave 1    0    0.00      0.00    0.00
   1              no religion 1634   37.87     40.61   40.61
   2                 buddhism   28    0.65      0.70   44.31
   3             christianity 1470   34.07     36.53   80.84
   4   christianity: catholic    0    0.00      0.00   80.84
   5 christianity: protestant    0    0.00      0.00   80.84
   6                 hinduism  103    2.39      2.56   83.40
   7                    islam  525   12.17     13.05   96.45
   8                  judaism    6    0.14      0.15   96.60
   9                  sikhism  137    3.17      3.40  100.00
  10           other religion  121    2.80      3.01   43.61
  NA                       NA  291    6.74        NA      NA
```

I don't want to see those `-88, -77, -66, -55, -1` in the output. Same for the plot. For example, 

`sjp.frq(en_w1$y1_rel1, coord.flip = TRUE, show.na = FALSE)
`
produces the following, 

![selection_279](https://user-images.githubusercontent.com/544033/40536616-14f9389e-600d-11e8-96a7-3c66dd9f1f54.png)

I don't want to see those "no answer", "not applicable" etc missing labels in the plot. Anyone here to help?

**Comment by haseebmahmud** [comment]
Wow! That was trivial. Thank you. 
BTW,  @strengejacke, thanks for the package. It makes our life lot easier.

**Comment by haseebmahmud** [comment]
Sorry, there is a problem I found. I am writing that now. Please do not close it.

**Comment by haseebmahmud** [comment]
I have a data of 4 different countries. I am importing those and then adding those together. I will explain what happens with and without `drop_labels()`.

First, let's see without `drop_labels()`. All the countries show a `frq` output like this,

```
 > frq(ge_w11$y1_rel1)

# m71: what is your religion (x) <categorical> 
# total N=5013  valid N=4883  mean=4.88  sd=2.04
 
 val                    label  frq raw.prc valid.prc cum.prc
 -88                no answer    0    0.00      0.00    0.00
 -77           not applicable    0    0.00      0.00    0.00
 -66       question not asked    0    0.00      0.00    0.00
 -55            other missing    0    0.00      0.00    0.00
  -1           dropout wave 1    0    0.00      0.00    0.00
   1              no religion  598   11.93     12.25   12.25
   2                 buddhism   20    0.40      0.41   16.28
   3             christianity    0    0.00      0.00   16.28
   4   christianity: catholic 1434   28.61     29.37   45.65
   5 christianity: protestant 1445   28.83     29.59   75.24
   6                 hinduism   17    0.34      0.35   75.59
   7                    islam 1182   23.58     24.21   99.80
   8                  judaism    7    0.14      0.14   99.94
   9                  sikhism    3    0.06      0.06  100.00
  10           other religion  177    3.53      3.62   15.87
  NA                       NA  130    2.59        NA      NA
```
Please note that, here we have labelled factors from 1 to 10 without any gap. 

Now if I include `drop_labels()`, then every country shows different output. For example, these are the output for all the countries,

```
> frq(en_w1$y1_rel1)

# m71: what is your religion (x) <categorical> 
# total N=4315  valid N=4024  mean=3.20  sd=2.58
 
 val          label  frq raw.prc valid.prc cum.prc
   1    no religion 1634   37.87     40.61   40.61
   2       buddhism   28    0.65      0.70   41.30
   3   christianity 1470   34.07     36.53   77.83
   6       hinduism  103    2.39      2.56   80.39
   7          islam  525   12.17     13.05   93.44
   8        judaism    6    0.14      0.15   93.59
   9        sikhism  137    3.17      3.40   96.99
  10 other religion  121    2.80      3.01  100.00
  NA             NA  291    6.74        NA      NA


> frq(ge_w1$y1_rel1)

# m71: what is your religion (x) <categorical> 
# total N=5013  valid N=4883  mean=4.88  sd=2.04
 
 val                    label  frq raw.prc valid.prc cum.prc
   1              no religion  598   11.93     12.25   12.25
   2                 buddhism   20    0.40      0.41   12.66
   4   christianity: catholic 1434   28.61     29.37   42.02
   5 christianity: protestant 1445   28.83     29.59   71.62
   6                 hinduism   17    0.34      0.35   71.96
   7                    islam 1182   23.58     24.21   96.17
   8                  judaism    7    0.14      0.14   96.31
   9                  sikhism    3    0.06      0.06   96.38
  10           other religion  177    3.53      3.62  100.00
  NA                       NA  130    2.59        NA      NA


> frq(nl_w1$y1_rel1)

# m71: what is your religion (x) <categorical> 
# total N=4363  valid N=4306  mean=3.29  sd=2.78
 
 val                    label  frq raw.prc valid.prc cum.prc
   1              no religion 2301   52.74     53.44   53.44
   2                 buddhism   17    0.39      0.39   53.83
   4   christianity: catholic  639   14.65     14.84   68.67
   5 christianity: protestant  391    8.96      9.08   77.75
   6                 hinduism   74    1.70      1.72   79.47
   7                    islam  648   14.85     15.05   94.52
   8                  judaism   18    0.41      0.42   94.94
   9                  sikhism    1    0.02      0.02   94.96
  10           other religion  217    4.97      5.04  100.00
  NA                       NA   57    1.31        NA      NA


> frq(sw_w1$y1_rel1)

# m71: what is your religion (x) <categorical> 
# total N=5025  valid N=4597  mean=3.19  sd=2.23
 
 val          label  frq raw.prc valid.prc cum.prc
   1    no religion 1444   28.74     31.41   31.41
   2       buddhism   52    1.03      1.13   32.54
   3   christianity 2211   44.00     48.10   80.64
   6       hinduism   13    0.26      0.28   80.92
   7          islam  779   15.50     16.95   97.87
   8        judaism    8    0.16      0.17   98.04
   9        sikhism    6    0.12      0.13   98.17
  10 other religion   84    1.67      1.83  100.00
  NA             NA  428    8.52        NA      NA
```

As you see, all 4 counties have some factors missing. E.g. in the first 4, 5, second country 3 and so on. Adding the 4 country together, I am getting 8 factors instead of 10. 

```
> all_plyr <- plyr::rbind.fill(en_w1, ge_w1, nl_w1, sw_w1)

> get_labels(all_plyr$y1_rel1)
[1] "no religion"    "buddhism"       "christianity"   "hinduism"       "islam"          "judaism"       
[7] "sikhism"        "other religion"
```
And I loose two factor labels here too as seen in the sjp.frq plot.

![2018-05-25 16_26_53-rstudio](https://user-images.githubusercontent.com/544033/40549740-8ae31f54-6038-11e8-96c7-beb0026316dc.png)

Any suggestion will be greatly appreciated.

**Comment by strengejacke** [comment]
could you please send me a small example dataset to reproduce this issue?

**Comment by haseebmahmud** [comment]
Here is one reproduction.

I have created two datasets, namely `data1` and `data2` having same coding schemes. I labeled the variable as color. The values are 1, 2 and 3. 1 is coded as Red, 2 is Blue and 3 is Green. Missing values are `-77, -88` and` -99`. 

In the first dataset, there is not 3 (Green). In the second dataset, all 3 colors are present. And here is a sample code to reproduce the error. 

```
library(sjPlot)
library(sjmisc)
library(sjlabelled)

data1 <- read_spss("data1.sav", atomic.to.fac = TRUE)
data2 <- read_spss("data2.sav", atomic.to.fac = TRUE)
all_plyr1 <- plyr::rbind.fill(data1, data2)

sjp.frq(all_plyr1$var1)

```
It looks fine. All the value labels are there.

![rplot1](https://user-images.githubusercontent.com/544033/40872288-fd910d5e-664b-11e8-9ed5-24aa7e61de71.png)

However, using `drop_labels()`, I have one label missing. 

```
data3 <- read_spss("data1.sav", atomic.to.fac = TRUE) %>% drop_labels()
data4 <- read_spss("data2.sav", atomic.to.fac = TRUE) %>% drop_labels()
all_plyr2 <- plyr::rbind.fill(data3, data4)

sjp.frq(all_plyr2$var1)
```
![rplot2](https://user-images.githubusercontent.com/544033/40872289-055091b8-664c-11e8-9fa8-0f9acc33102b.png)

The demo datasets and the code can be found [here](https://github.com/haseebmahmud/sjPlot_error_demo).
